scalar_test.go

```diff
@@ -40,9 +40,9 @@ func TestMarshalID(t *testing.T) {
 	buf3, err3 := json.Marshal(case3)
 	// Assert
 	autopilot.Ok(t, err1)
-	autopilot.Equals(t, `{"key1":"","key3":null}`, string(buf1))
+	autopilot.Equals(t, `{"key1":null,"key3":null}`, string(buf1))
```

Comment:
null for IDs means "unset" in our API generally but what does `""` mean?  I'm concerned we are losing the ability to to create input types where `""` is the expected JSON.

What is the underlying impetus for this change?  golang has the ability to put functions on pointers or value types - theres no reason that I see restricting this.